### DISCUSSION TITLE
Xdrip and Libre 3. XDrip doesn't get sugars from Web Follower

### DISCUSSION THREAD

**Discussion by guestfreeman** [NightscoutFoundation/xDrip#3041] [Q&A]
Hi everyone. There's a 1st phone with XDrip (23ccbd1-2023.07.16) and FreestyleLibre 3 installed. Freestyle Libre 3 has account registered.  There's a 2nd phone with LibreLinkup, registered account. The 2nd phone gets sugar from the 1st phone. In Xdrip, the source is Web Follower, e-mail and password are written from the accound for LibreLinkUp on the 2nd phone. 
The problem. XDrip doesn't get sugar from LIbre.  There is an error on System Status - Web Follower Status. "Pprocess failure status-AZ {"status":2,"error":{"message":"notAuthenticated"}}.  I'm sure that I've written the email and the password correctrly. How to resolve the problem ?

**Comment by Navid200** [comment]
Follow this and tell us what happens.

https://navid200.github.io/xDrip/docs/Libre3_Follow.html

**Reply by guestfreeman** [reply]
Hi. I've done it. There`re nothing. In the Web Follower status - no any arrors.

**Reply by guestfreeman** [reply]
I've reinstallrd XDrip, re—typed email, password, and nothing.

**Reply by PhilTU** [reply]
Try a different email.?
I had a similar problem. Web Follower never collected data.

**Reply by Navid200** [reply]
You may be causing this problem yourself.
Read your first post.  Why are there two phones involved?
Can you get everything working as it is described in the link I provided.  It doesn't talk about two different phones.
You are supposed to install and have the official app working on the same phone you run xDrip on.  Can you do that?

**Reply by PhilTU** [reply]
I think they have LibrelinkUp app and xDrip both on phone 2?

To be fair the instructions don't specify the use of a single phone.

**Comment by guestfreeman** [comment]
I've tried a different e-mail. In the "Web Follower Status" - Last reading: n/a. No any errors. I think there are buhs in XDrip...
May be those bugs should be fixed ?

**Reply by PhilTU** [reply]
no relevant messages in Errors / events log ?

I agree it needs fixing, although I am not a fan of these arrangements to follow vendor cloud systems when the vendor is actively changing things to prevent them working. Seems like a sort of arms race.

If we have error messages someone may take a look, perhaps more logging is required to be turned on but I didn't see anything useful when it failed to collect for me.

**Reply by guestfreeman** [reply]
Hi. Where should I get event logs ? 
There are no any errors in System Status - Web follower, just "Latest reading: n/a", without any errors.

**Reply by guestfreeman** [reply]
Main screen - >right top corner->3dots-> Event log view, there are events:
1. Jamorham GCMils: "We don't have any more recent data to offer than: no data. 3 events
2. WebFollow: Failed to get any responce. 5 events
3. Again Jamorham GCMils, 1 events
4. WebFollow. 1 event
5. Jamorham GCMils. 1 events
6. etc.

**Reply by PhilTU** [reply]
Failed to get any response is interesting. Librelinkup not responding. Hmm.

**Reply by jrdeutsch** [reply]
The Libre Follower app needs to be on the same phone as the main Libre app as well as the XDrip+ app. All 3 apps on the same phone.

**Reply by PhilTU** [reply]
Interesting, as LibreLinkUp app is clearly intended to be on a different phone. G-watch wear will collected data on a different phone too, so is there a specific design feateure of xDrip+ Web Follow that requires this ?

**Reply by Navid200** [reply]
According to the instructions, after everything is working, you can uninstall the Link-up app.
https://navid200.github.io/xDrip/docs/Libre3_Follow.html

**Comment by tqkam** [comment]
Hi,
I have found, that the custom script server https://community-script.beonlabs.net/ returns a file with a message "service ceased".
It looks like the source isn't supported more.
